plugins/obs-qsv11/common_utils_linux.cpp

```diff
@@ -0,0 +1,274 @@
+	UNUSED_PARAMETER(lock_key);
+	UNUSED_PARAMETER(next_key);
+	return MFX_ERR_UNSUPPORTED;
+}
+
+#if 0
+void ClearYUVSurfaceVMem(mfxMemId memId);
+void ClearRGBSurfaceVMem(mfxMemId memId);
+#endif
+
+// Initialize Intel Media SDK Session, device/display and memory manager
+mfxStatus Initialize(mfxIMPL impl, mfxVersion ver, MFXVideoSession *pSession,
+		     mfxFrameAllocator *pmfxAllocator, mfxHDL *deviceHandle,
+		     bool bCreateSharedHandles, bool dx9hack)
+{
+	UNUSED_PARAMETER(pmfxAllocator);
+	UNUSED_PARAMETER(deviceHandle);
+	UNUSED_PARAMETER(bCreateSharedHandles);
+	UNUSED_PARAMETER(dx9hack);
+	mfxStatus sts = MFX_ERR_NONE;
+
+	// Initialize Intel Media SDK Session
+	sts = pSession->Init(impl, &ver);
+	MSDK_CHECK_RESULT(sts, MFX_ERR_NONE, sts);
+	return sts;
+}
+
+// Release resources (device/display)
+void Release(){};
+
+void mfxGetTime(mfxTime *timestamp)
+{
+	clock_gettime(CLOCK_MONOTONIC, timestamp);
+}
+
+//void mfxInitTime();  might need this for Windows
+double TimeDiffMsec(mfxTime tfinish, mfxTime tstart)
+{
+	UNUSED_PARAMETER(tfinish);
+	UNUSED_PARAMETER(tstart);
+	//TODO, unused so far it seems
+	return 0.0;
+}
+
+extern "C" void util_cpuid(int cpuinfo[4], int level)
+{
+	__get_cpuid(level, (unsigned int *)&cpuinfo[0],
+		    (unsigned int *)&cpuinfo[1], (unsigned int *)&cpuinfo[2],
+		    (unsigned int *)&cpuinfo[3]);
+}
+
+struct vaapi_device {
+	int fd;
+	VADisplay display;
+	const char *driver;
+};
+
+void vaapi_open(char *device_path, struct vaapi_device *device)
+{
+	int fd = open(device_path, O_RDWR);
+	if (fd < 0) {
+		return;
+	}
+
+	VADisplay display = vaGetDisplayDRM(fd);
+	if (!display) {
+		close(fd);
+		return;
+	}
+
+	// VA-API is noisy by default.
+	vaSetInfoCallback(display, nullptr, nullptr);
+	vaSetErrorCallback(display, nullptr, nullptr);
+
+	int major, minor;
+	if (vaInitialize(display, &major, &minor) != VA_STATUS_SUCCESS) {
+		vaTerminate(display);
+		close(fd);
+		return;
+	}
+
+	const char *driver = vaQueryVendorString(display);
+	if (strstr(driver, "Intel i965 driver") != nullptr) {
+		blog(LOG_ERROR,
+		     "Legacy intel-vaapi-driver detected, incompatible with QSV");
+		vaTerminate(display);
+		close(fd);
+		return;
+	}
+
+	device->fd = fd;
+	device->display = display;
+	device->driver = driver;
+}
+
+void vaapi_close(struct vaapi_device *device)
+{
+	vaTerminate(device->display);
+	close(device->fd);
+}
+
+static uint32_t vaapi_check_support(VADisplay display, VAProfile profile,
+				    VAEntrypoint entrypoint)
+{
+	bool ret = false;
+	VAConfigAttrib attrib[1];
+	attrib->type = VAConfigAttribRateControl;
+
+	VAStatus va_status =
+		vaGetConfigAttributes(display, profile, entrypoint, attrib, 1);
+
+	uint32_t rc = 0;
+	switch (va_status) {
+	case VA_STATUS_SUCCESS:
+		rc = attrib->value;
+		break;
+	case VA_STATUS_ERROR_UNSUPPORTED_PROFILE:
+	case VA_STATUS_ERROR_UNSUPPORTED_ENTRYPOINT:
+	default:
+		break;
+	}
+
+	return (rc & VA_RC_CBR || rc & VA_RC_CQP || rc & VA_RC_VBR);
```

Comment:
Another idea:
Make sure that the adapter handle the three rate control, which allow to keep QSV in simple mode on Linux since "problematic" CPU are removed from the equation.
```suggestion
	return (rc & VA_RC_CBR && rc & VA_RC_CQP && rc & VA_RC_VBR);
```

Reply:
This is being used for support of the encoder in general, setting it this way would just disable support entirely on free drivers. I think it probably makes sense to not have the encoder in streaming if it doesnt support CBR, but I would prefer to keep it available for recording even in that case.

Reply:
(and also the h264 version of this function isnt being used at all, since the shared windows code just assumes h264 support)